geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/operations/GetAllRequestOperationHandler.java

```diff
@@ -61,41 +58,40 @@
     }
 
     long startTime = messageExecutionContext.getStatistics().startOperation();
-    Map<Boolean, List<Object>> resultsCollection;
+    RegionAPI.GetAllResponse.Builder responseBuilder = RegionAPI.GetAllResponse.newBuilder();
     try {
-      ((InternalCache) messageExecutionContext.getCache()).setReadSerializedForCurrentThread(true);
+      messageExecutionContext.getCache().setReadSerializedForCurrentThread(true);
+
+      for (BasicTypes.EncodedValue key : request.getKeyList()) {
+        Object entry = processOneMessage(serializationService, region, key);
+        if (entry instanceof BasicTypes.Entry) {
+          responseBuilder.addEntries((BasicTypes.Entry) entry);
+        } else {
+          responseBuilder.addFailures((BasicTypes.KeyedError) entry);
+        }
+      }
 
-      resultsCollection = request.getKeyList().stream()
-          .map((key) -> processOneMessage(serializationService, region, key))
-          .collect(Collectors.partitioningBy(x -> x instanceof BasicTypes.Entry));
     } finally {
-      ((InternalCache) messageExecutionContext.getCache()).setReadSerializedForCurrentThread(false);
+      messageExecutionContext.getCache().setReadSerializedForCurrentThread(false);
       messageExecutionContext.getStatistics().endOperation(startTime);
     }
-    RegionAPI.GetAllResponse.Builder responseBuilder = RegionAPI.GetAllResponse.newBuilder();
-
-    for (Object entry : resultsCollection.get(true)) {
-      responseBuilder.addEntries((BasicTypes.Entry) entry);
-    }
-
-    for (Object entry : resultsCollection.get(false)) {
-      responseBuilder.addFailures((BasicTypes.KeyedError) entry);
-    }
 
     return Success.of(responseBuilder.build());
   }
 
   private Object processOneMessage(ProtobufSerializationService serializationService, Region region,
-      BasicTypes.EncodedValue key) {
+      BasicTypes.EncodedValue key) throws DecodingException {
     try {
       Object decodedKey = serializationService.decode(key);
       Object value = region.get(decodedKey);
       return ProtobufUtilities.createEntry(serializationService, decodedKey, value);
     } catch (EncodingException ex) {
       logger.error("Encoding not supported: {}", ex);
       return createKeyedError(key, "Encoding not supported.", INVALID_REQUEST);
+    } catch (DecodingException ex) {
```

Comment:
So this will now result in failing the entire operation when a exception occurs during decoding rather than just failing the key we failed to decode.  Is that intended, and is it also intended that encoding exceptions will continue to follow the old behavior of only failing the single operation?

Reply:
Hmm, sounds like we need  a test for that.  I'll write one and fix this to not throw DecodingException.